Improvement: publishing scripts (#347)
codemagic-ci-cd/codemagic-docs: content/getting-started/yaml.md

### Publishing

Codemagic has a number of integrations (e.g. email, Slack, Google Play, App Store Connect) for publishing but you can also publish elsewhere with custom scripts (e.g. Firebase App Distribution). See the examples [here](../publishing-yaml/distribution).

Note that unlike the scripts the scripts section, the scripts in publishing scripts section are run regardless of the build status. You can specify conditions for running the scripts. For example, you can check whether artifacts were generated or what the build status was.

```yaml
publishing:
  email:
    recipients:
      - name@example.com
  scripts:
    script: |
      if ls /path/to/artifacts/**/*.ipa 1> /dev/null 2>&1; then
        # publish artifacts
      fi 
```

> 

Reviewer: This requires some explanation here, e.g.:
You can also use the post-publishing scripts to report build status.

  email:
    recipients:
      - name@example.com
  scripts:
    - |
      echo 'This is a Post-publish script'
      echo 'This script is multiline'
```

Add custom scripts to call different actions depending on final build status. See the example below

```yaml
scripts:
  - name: Report build start
    script: # build started

    . . .

  - name: Build finished successfully
    script: touch ~/SUCCESS
publishing:
  scripts:
    - name: Report build status
      script: |
        if [ -a "~/SUCCESS" ] ; then
           # build successful
        else
           # build failed
        fi
```

> Note that in this example the `SUCCESS` file will not be created if some build step failed. We it's only used to distinguish between failed and successfull builds. 

Reviewer: ```suggestion
Note that in this example the `SUCCESS` file will not be created if some build step failed. It is only used to distinguish between failed and successful builds. 
```

> Unlike the scripts run in the scripts section before the publishing step, the post-publish scripts are run regardless of the status of previous scripts.

Reviewer: let's move this right after the intro paragraph for ### Publishing

> Unlike the scripts run in the scripts section before the publishing step, the post-publish scripts are run regardless of the status of previous scripts.

Reviewer: ```suggestion
Unlike the scripts run in the scripts section before the publishing step, the publishing scripts are run regardless of the status of previous scripts.
```
